File: Anya_Seth/2-Pointer/Reverse Vowels of a String.java
class Solution {
    public String reverseVowels(String s) {
       int i = 0;
       int j = s.length() - 1;
       char[] arr = s.toCharArray();
       while(i < j){
        while(i < j && !isVowels(arr[i])){
            i++;
        }
        while(i < j && !isVowels(arr[j])){
            j--;
        }
        if(i < j){
             char temp = arr[i];
               arr[i] = arr[j];
               arr[j] = temp;
               i++;
               j--;
        }
        }
        return new String(arr); 
    }
    public static boolean isVowels(char x) 
    { 
        if (x == 'a' || x == 'e' || x == 'i' || x == 'o' || x == 'u' || x == 'A' || 
            x == 'E' || x == 'I' || x == 'O' || x == 'U'){
            return true;
        }
        return false;
    } 
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Reverse_Vowels_of_a_String.cpp
class Solution {
public:
    bool isvow(char c){
        if(c=='a' || c=='e' || c=='i' || c=='o' || c=='u' || c=='A' || c=='E' || c=='I' || c=='O' || c=='U'){
            return true;
        }
        return false;
    }
    string reverseVowels(string s) {
        int n=s.size();
        int i=0,j=n-1;
        while(i<j){
            if(isvow(s[i]) && isvow(s[j])){
                swap(s[i],s[j]);
                i++;
                j--;
                continue;
            }
            if(isvow(s[i]) && !isvow(s[j])){
                j--;
                continue;
            }
            if(!isvow(s[i]) && isvow(s[j])){
                i++;
                continue;
            }          
            i++;
            j--;       
        }
        return s;
    }
};
